src/app/ClusterData.h

```diff
@@ -0,0 +1,382 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    Copyright (c) 2018 Google LLC.
+ *    Copyright (c) 2016-2017 Nest Labs, Inc.
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file forms the crux of the Cluster Data Management layer, providing
+ *      various classes that manage and process data as it applies to Clusters and their
+ *      associated schemas.
+ *
+ */
+
+#pragma once
+
+#include <app/util/basic-types.h>
+#include <core/CHIPCore.h>
+#include <core/CHIPTLVDebug.hpp>
+#include <lib/core/CHIPError.h>
+
+namespace chip {
+namespace app {
+typedef uint32_t AttributePathHandle;
+typedef uint32_t ClusterDataHandle;
+/**
+ * A AttributePathHandle is a unique 32-bit numerical hash of a IM path relative to the root of a Cluster instance. It has two parts
+ * to it:
+ *
+ * Some characteristics:
+ *  - Every Cluster has its own Attribute path handle space.
+ *  - Every unique IM attribute sub-path path will have a similarly unique AttributePathHandle.
+ *  - With this construct, consumer SDK logic never has to deal with IM paths directly. Rather, their interactions with IM are
+ *    conducted exclusively through these handles.
+ *
+ */
+
+// 0 indicates a 'NULL' handle
+constexpr AttributePathHandle kNullAttributePathHandle = 0;
+// 1 indicates a handle that points to the root of the Cluster instance.
+constexpr AttributePathHandle kRootAttributePathHandle = 1;
+/* Offset from the value of a path that provides the index into the table for that handle
+ * It is 2 since the 0 and 1 are special handles.
+ */
+constexpr AttributePathHandle kHandleTableOffset = 2;
+
+inline bool IsRootAttributePathHandle(AttributePathHandle aHandle)
```

Comment:
`IsRootAttributePathHandle(handle)` seems just as readable as `handle == kRootAttributePathHandle`.

Please remove the extra indirection here and for null attribute path handle.